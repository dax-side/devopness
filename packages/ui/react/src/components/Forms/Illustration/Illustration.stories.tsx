import type { Meta, StoryObj } from '@storybook/react-vite'

import { Illustration } from './Illustration'
import { getColor } from 'src/colors'
import { getImageAssetUrl } from 'src/icons'
const EMPTY_APPLICATIONS = getImageAssetUrl('applications_module.svg')

const meta = {
  title: 'Form/Illustration',
  component: Illustration,
  parameters: {
    backgrounds: {
      options: {
        purple: { name: 'Purple', value: getColor('purple.800') },
      },
    },
  },
  globals: {
    backgrounds: { value: 'purple' },
  },
} satisfies Meta<typeof Illustration>

type Story = StoryObj<typeof meta>

const Default: Story = {
  args: {
    children: (
      <img
        src={EMPTY_APPLICATIONS}
        alt="placeholder"
      />
    ),
  },
}

const WithText: Story = {
  args: {
    children: (
      <p style={{ fontSize: '18px' }}>Centered text inside Illustration</p>
    ),
  },
}

const WithCustomContent: Story = {
  args: {
    children: null,
  },
  render: () => (
    <Illustration>
      <div
        style={{
          width: 60,
          height: 60,
          background: getColor('blue.600'),
          borderRadius: '50%',
        }}
      />
    </Illustration>
  ),
}

export { Default, WithText, WithCustomContent }
export default meta
