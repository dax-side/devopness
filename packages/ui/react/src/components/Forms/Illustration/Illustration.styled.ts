import { styled } from 'styled-components'

import { getColor } from 'src/colors'

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  width: 100%;
  height: 126px;
  border-bottom: 1px solid ${getColor('slate.300')};
`
export { Wrapper }
