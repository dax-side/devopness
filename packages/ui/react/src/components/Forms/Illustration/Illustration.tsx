import React from 'react'

import { Wrapper } from './Illustration.styled'

/**
 * Illustration Component
 *
 * A simple flex container to center content horizontally and vertically,
 * with a fixed height and bottom border.
 *
 * @example
 * ```tsx
 * <Illustration>
 *   <img src="logo.png" alt="Logo" />
 * </Illustration>
 * ```
 */
type IllustrationProps = {
  /** React children to render inside */
  children: React.ReactNode
}

const Illustration = ({ children }: IllustrationProps) => (
  <Wrapper>{children}</Wrapper>
)

export { Illustration }
export type { IllustrationProps }
