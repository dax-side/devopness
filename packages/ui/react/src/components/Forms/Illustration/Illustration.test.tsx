import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'

import { Illustration } from './Illustration'

describe('Illustration', () => {
  it('renders children correctly', () => {
    render(
      <Illustration>
        <p data-testid="child">Hello</p>
      </Illustration>
    )
    expect(screen.getByTestId('child')).toBeInTheDocument()
    expect(screen.getByText('Hello')).toBeVisible()
  })

  it('applies flex centering styles', () => {
    render(
      <Illustration>
        <span>Content</span>
      </Illustration>
    )
    const wrapper = screen.getByText('Content').parentElement
    expect(wrapper).toHaveStyle({
      display: 'flex',
      'justify-content': 'center',
      'align-items': 'center',
    })
  })

  it('has fixed height and border bottom', () => {
    const { container } = render(
      <Illustration>
        <span>Border test</span>
      </Illustration>
    )
    const wrapper = screen.getByText('Border test').parentElement

    expect(wrapper).toHaveStyle({
      height: '126px',
      width: '100%',
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
