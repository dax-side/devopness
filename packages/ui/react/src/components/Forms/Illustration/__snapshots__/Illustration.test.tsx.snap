// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Illustration > has fixed height and border bottom 1`] = `
<div
  class="sc-bRKDuR laxXAR"
>
  <span>
    Border test
  </span>
</div>
`;
